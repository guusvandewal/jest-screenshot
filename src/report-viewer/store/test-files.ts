import { component } from "tsdi";
import { observable, action, computed } from "mobx";
import { FileReport } from "../../reporter-types";

@component
export class StoreTestFiles {
    @observable public activeFileName = testResults.files.length > 0 ? testResults.files[0].testFilePath : undefined;

    @action.bound public selectFile(file: FileReport) {
        this.activeFileName = file.testFilePath;
    }

    @computed public get activeFile() {
        return testResults.files.find(file => file.testFilePath === this.activeFileName);
    }

    public isActive(file: FileReport) {
        return this.activeFileName === file.testFilePath;
    }
}
