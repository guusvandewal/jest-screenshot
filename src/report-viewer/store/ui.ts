import { component } from "tsdi";
import { observable, action, computed } from "mobx";
import { FileReport } from "../../reporter-types";
import { bind } from "lodash-decorators";

@component
export class StoreUi {
    @observable public menuVisible = true;

    @bind @action public toggleMenu() {
        this.menuVisible = !this.menuVisible;
    }
}
