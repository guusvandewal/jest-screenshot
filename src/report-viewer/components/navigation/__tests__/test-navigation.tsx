import * as React from "react";
import { shallow } from "enzyme";
import { Navigation } from "..";
import { StoreUi } from "../../../store";

describe("Navigation", () => {
    it("looks as expected", () => {
        expect(shallow(<Navigation />)).toMatchSnapshot();
    });

    it("clicking on the menu toggle button", () => {
        const element = shallow(<Navigation />);
        expect(tsdi.get(StoreUi).menuVisible).toBe(true);
        element.find("a.navbar-item").simulate("click");
        expect(tsdi.get(StoreUi).menuVisible).toBe(false);
    });
});
