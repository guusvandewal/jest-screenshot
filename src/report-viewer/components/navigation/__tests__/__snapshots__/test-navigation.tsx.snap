// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation looks as expected 1`] = `
<nav
  className="navbar is-dark"
>
  <div
    className="navbar-brand brand"
  >
    Jest Screenshot Report
  </div>
  <div
    className="navbar-menu is-active"
  >
    <div
      className="navbar-end"
    >
      <a
        className="navbar-item"
        onClick={[Function]}
      >
        Hide Sidebar
      </a>
    </div>
  </div>
</nav>
`;
